stat/stat.go

```diff
@@ -1422,6 +1426,50 @@ func linInterpQuantile(p float64, x, weights []float64, sumWeights float64) floa
 	panic("impossible")
 }
 
+func linearQuantile(p float64, x, weights []float64, sumWeights float64) float64 {
+	n := len(x)
+
+	// Unweighted: Hyndman & Fan Type 7 (R default).
+	if weights == nil {
+		if p <= 0 {
+			return x[0]
+		}
+		if p >= 1 {
+			return x[n-1]
+		}
+		h, frac := math.Modf(1 + p*float64(n-1))
+		i := int(h) - 1
+		if frac == 0 {
+			return x[i]
+		}
+		return (1-frac)*x[i] + frac*x[i+1]
+	}
+
+	// Weighted: linear interpolation over the weighted ECDF.
+	// On exact boundaries, return the current x[i] (no averaging).
+	fidx := p * sumWeights
+	var cumsum float64
+	for i := range x {
+		w := weights[i]
+		prev := cumsum
+		cumsum += w
+		if cumsum >= fidx {
+			if i == 0 {
+				return x[0]
+			}
+			alpha := (fidx - prev) / w // in [0,1]
+			if alpha <= 0 {
+				return x[i-1]
+			}
+			if alpha >= 1 {
+				return x[i]
+			}
+			return (1-alpha)*x[i-1] + alpha*x[i]
```

Comment:
```suggestion
			return float64((1-alpha)*x[i-1]) + float64(alpha*x[i])
```
to avoid FMA which can cause differences in behaviour between arm64 and amd64.

Reply:
Amended ✅

Reply:
I'd also push back on the idea that we need to do this altogether as alpha and x[i] are both float64.

Reply:
But they’re already a float64 type …  we don’t need to do any further type conversion. It looks to me that this is redundant in this situation and perhaps a misunderstanding of the above specification you linked. The tests passing both before and after perhaps suggest the change is not necessary.

Reply:
Yeh but I certainly wouldn’t advocate for increasing tolerances. Happy to close this out and leave as is to ensure perfect precision across multiple architectures.